src/Analyser/NodeScopeResolver.php

```diff
@@ -2936,12 +2940,16 @@ static function (Node $node, Scope $scope) use ($nodeCallback): void {
 			$throwPoints = [];
 		}
 
+		if ($falsyScope === null) {
+			$falsyScope = $scope;
```

Comment:
It'd make more sense to just do `return new ExpressionResult(` in the `Isset_` branch instead of declare and use a variable thousands of lines from each other.